evalml/model_understanding/graphs.py

```diff
@@ -436,6 +436,12 @@ def partial_dependence(pipeline, X, feature, grid_resolution=100):
     elif isinstance(pipeline, evalml.pipelines.RegressionPipeline):
         pipeline._estimator_type = "regressor"
     pipeline.feature_importances_ = pipeline.feature_importance
+    if isinstance(feature, int):
```

Comment:
I think NaNs would still be an issue if they are present in another feature and not in the feature of interest for the partial dependence. I wonder if it's better to issue the warning if the pipeline does not have an imputer (something like `Imputer not present in pipeline, partial_dependence may have NaN values if NaN values are present in the data`)? 

I'm guessing NaNs would throw off 95% of our features and it's not sustainable to independently check for NaNs in all places. I think it's on the user to add an imputer to the pipeline if nans are possible in their data. Maybe the question should be whether or not that's communicated well in the docs. What do you think?